neoexchange/astrometrics/tests/test_sources_subs.py

```diff
@@ -619,6 +650,69 @@ def test_target_with_ampersand4(self):
         self.assertEqual(1, len(targets))
         self.assertEqual(expected_target, targets)
 
+    @freeze_time(datetime(2021, 12, 7, 2, 0, 0))
+    def test_csv_file_2021(self):
+
+        expected_targets = ['163899', '4660', '2021 XK6']
+
+        targets = fetch_goldstone_targets(self.test_csv_file)
+
+        self.assertEqual(3, len(targets))
+        self.assertEqual(expected_targets, targets)
+
+    @freeze_time(datetime(2022, 1, 7, 2, 0, 0))
+    def test_csv_file_2022(self):
+
+        expected_targets = ['7842', '153591', '2016 QJ44', '2018 CW2', '2010 XC15']
+
+        targets = fetch_goldstone_targets(self.test_csv_file)
+
+        self.assertEqual(5, len(targets))
+        self.assertEqual(expected_targets, targets)
+
+    @freeze_time(datetime(2021, 12, 7, 2, 0, 0))
+    def test_csv_file_calformat_2021(self):
+
+        expected_targets = ['163899', '4660', '2021 XK6']
+
+        expected_targets = [{ 'target': '163899',
+                              'windows': [{'start': '2021-11-22T00:00:00', 'end': '2021-12-31T23:59:59'}]},
+                             {'target': '4660',
+                              'windows': [{'start': '2021-12-05T00:00:00', 'end': '2021-12-31T23:59:59'}]},
+                             {'target': '2021 XK6',
+                              'windows': [{'start': '2021-12-17T00:00:00', 'end': '2021-12-17T23:59:59'}]},
+                              ]
+
+
+        targets = fetch_goldstone_targets(self.test_csv_file, calendar_format=True)
+
+        self.assertEqual(3, len(targets))
+        self.assertEqual(expected_targets, targets)
+
+    @freeze_time(datetime(2022, 1, 7, 2, 0, 0))
+    def test_csv_file_calformat_2022(self):
+
+        expected_targets = ['7842', '153591', '2016 QJ44', '2018 CW2', '2010 XC15']
+
+        expected_targets = [
+                             {'target': '7842',
+                              'windows': [{'start': '2022-01-18T00:00:00', 'end': '2022-01-25T23:59:59'}]},
+                             {'target': '153591',
+                              'windows': [{'start': '2022-02-18T00:00:00', 'end': '2022-03-08T23:59:59'}]},
+                             {'target': '2016 QJ44',
+                              'windows': [{'start': '2022-02-18T00:00:00', 'end': '2022-02-25T23:59:59'}]},
+                             {'target': '2018 CW2',
+                              'windows': [{'start': '2022-02-16T00:00:00', 'end': '2022-02-21T23:59:59'}]},
+                             {'target': '2010 XC15',
+                              'windows': [{'start': '2022-12-24T00:00:00', 'end': '2023-01-06T23:59:59'}]},
+                              ]
+
+
+        targets = fetch_goldstone_targets(self.test_csv_file, calendar_format=True)
+
+        self.assertEqual(len(expected_targets), len(targets))
```

Comment:
This test seems unnecessary given the next one...

Reply:
It's cleaner to see a basic "is there the correct number of entries" test first; the list of JSON dictionaries gets a bit long and hard to parse in the test output when they differ

Reply:
fair enough.